src/Model/DataModel/StreamData/HoldEmailData.php

```diff
@@ -0,0 +1,152 @@
+<?php
+namespace NYPL\HoldRequestResultConsumer\Model\DataModel\StreamData;
+
+use NYPL\HoldRequestResultConsumer\Model\DataModel\StreamData;
+
+class HoldEmailData extends StreamData
+{
+    /**
+     * @var string
+     */
+    public $patronName = '';
+
+    /**
+     * @var string
+     */
+    public $title = '';
+
+    /**
+     * @var string
+     */
+    public $author = '';
+
+    /**
+     * @var string
+     */
+    public $barcode = '';
+
+    /**
+     * @var string
+     */
+    public $pickupLocation = '';
+
+    /**
+     * @var string
+     */
+    public $deliveryLocation = '';
+
+    /**
+     * HoldEmailData constructor.
+     * @param Patron $patron
+     * @param Bib $bib
+     * @param Item $item
+     * @param HoldRequest $holdRequest
+     */
+    public function __construct(Patron $patron, Bib $bib, Item $item, HoldRequest $holdRequest)
```

Comment:
The constructor is overridden because the data source for this object comes from several data objects. This is the most efficient way to generate this object at the moment.